### PR TITLE
Improve logging in gRPC server methods

### LINKED ISSUE #384: Improve Logging in gRPC Server methods
# Summary

Currently the logging inside the various gRPC method implementations is all over the place. We should harmonize and add meaningful logging.

## Expected outcome

- It should be clear in which context/volume/bucket a certain log entry has been created
- The flow of operations must be clear from the logging: e.g. "Creating ...." at the beginning and "Created ...." to indicate when certain operations started and ended.

### PR DESCRIPTION
**Pull request by sujeet01**
Fixes #384

### CODE REVIEW

File: ori/bucket/server/bucket_create.go [hunk 1]
```diff
@@ -53,12 +53,12 @@ func (s *Server) createBucketClaimAndAccessSecretFromBucket(
 	apiutils.SetClassLabel(bucketClaim, bucket.Spec.Class)
 	apiutils.SetBucketManagerLabel(bucketClaim, apiutils.BucketManager)
 
-	log.V(1).Info("Creating bucket claim")
+	log.V(2).Info("Creating bucket claim")
 	if err := s.client.Create(ctx, bucketClaim); err != nil {
-		return nil, nil, fmt.Errorf("error creating bucket: %w", err)
+		return nil, nil, fmt.Errorf("error creating bucket claim: %w", err)
```
Comment (lukasfrank) [inline ori/bucket/server/bucket_create.go:58]: "failed to create bucket claim:"

File: ori/bucket/server/bucket_create.go [hunk 2]
```diff
@@ -72,19 +72,21 @@ func (s *Server) CreateBucket(
...
 
-	log.V(1).Info("Create bucket claim and bucket access secret")
+	log.V(1).Info("Creating bucket claim and bucket access secret")
 	bucketClaim, accessSecret, err := s.createBucketClaimAndAccessSecretFromBucket(ctx, log, req.Bucket)
 	if err != nil {
 		return nil, fmt.Errorf("error getting bucket config: %w", err)
 	}
 
-	log.V(1).Info("Generate ORI bucket object")
+	log.V(1).Info("Generating ORI bucket object")
```
Comment (lukasfrank) [inline ori/bucket/server/bucket_create.go:83]: "Getting"

File: ori/bucket/server/bucket_list.go [hunk 3]
```diff
@@ -149,6 +149,10 @@ func (s *Server) getBucketForID(ctx context.Context, id string) (*oriv1alpha1.Bu
 }
 
 func (s *Server) ListBuckets(ctx context.Context, req *oriv1alpha1.ListBucketsRequest) (*oriv1alpha1.ListBucketsResponse, error) {
+	log := s.loggerFrom(ctx)
+
+	log.V(1).Info("Listing buckets", "args", *req)
```
Comment (lukasfrank) [inline ori/bucket/server/bucket_list.go:154]: I would propose to decrease the log level, if we would like to keep this log line. I think this information is not needed on every call of the ListBuckets endpoint.

File: ori/volume/server/volume_list.go [hunk 4]
```diff
@@ -88,6 +88,8 @@ func (s *Server) listVolumes(ctx context.Context, log logr.Logger) ([]*ori.Volum
 func (s *Server) ListVolumes(ctx context.Context, req *ori.ListVolumesRequest) (*ori.ListVolumesResponse, error) {
 	log := s.loggerFrom(ctx)
 
+	log.V(1).Info("Listing volumes", "args", *req)
```
Comment (lukasfrank) [inline ori/volume/server/volume_list.go:91]: Same here

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: ori/bucket/server/bucket_list.go (hunk 3 above)
Comment (sujeet01) [reply ori/bucket/server/bucket_list.go]: Done

File: ori/bucket/server/bucket_list.go [hunk 5]
```diff
@@ -149,6 +149,10 @@ func (s *Server) getBucketForID(ctx context.Context, id string) (*oriv1alpha1.Bu
 }
 
 func (s *Server) ListBuckets(ctx context.Context, req *oriv1alpha1.ListBucketsRequest) (*oriv1alpha1.ListBucketsResponse, error) {
+	log := s.loggerFrom(ctx)
+
+	log.V(2).Info("Listing buckets", "args", *req)
```
Comment (afritzler) [inline ori/bucket/server/bucket_list.go:154]:
```suggestion
	log.V(2).Info("Listing buckets")
```

Comment (afritzler) [reply ori/bucket/server/bucket_list.go:154]: This will produce a messy output.

File: ori/volume/server/volume_create.go [hunk 6]
```diff
@@ -80,36 +81,40 @@ func (s *Server) createImageFromVolume(ctx context.Context, log logr.Logger, vol
...
 		return nil, fmt.Errorf("failed to create image: %w", err)
 	}
 
+	log.V(2).Info("Image created", "imageID", image.Metadata.ID)
 	return image, nil
 }
 
 func (s *Server) CreateVolume(ctx context.Context, req *ori.CreateVolumeRequest) (res *ori.CreateVolumeResponse, retErr error) {
 	log := s.loggerFrom(ctx)
+	log.V(1).Info("Creating volume", "args", *req)
```
Comment (afritzler) [inline ori/volume/server/volume_create.go:103]: Can we set the logger context to use the image ID? That way we don't have to log the complete request struct.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: ori/volume/server/volume_create.go (hunk 6 above)
Comment (sujeet01) [reply ori/volume/server/volume_create.go]: Done

**Review by lukasfrank — APPROVED** [review]

**Pull request merged**